### DISCUSSION TITLE
Is it a good idea to call multiple request from single endpoint?

### DISCUSSION THREAD

**Discussion by berubejp** [LuckyPennySoftware/MediatR#889] [General]
Hi,

I've got a CRUD web api with code managed as features. So I've got GetAll, GetById, Create, Update and Delete features with corresponding requests/handlers.

When I create or update my entity, I need to return my entity, which is the same code/feature as GetById. 

So my question, is it a good idea to send multiple request from a single endpoint?

Here is what would be the result in my controller.

```csharp
[HttpGet("{id}")]
public async Task<IActionResult> Get([FromRoute]int id)
{
    var entite = await _mediator.Send(new GetByIdEntityRequest { Id = id });
    return Ok(entite );
}

[HttpPost]
public async Task<IActionResult> Post([FromBody]CreateEntityRequest request)
{
    var id = await _mediator.Send(request); // Create entity
    var entite = await _mediator.Send(new GetByIdEntityRequest { Id = id }); // Get returned dto
    return CreatedAtAction(nameof(Get), new { entite.Id }, entite);
}

[HttpPut]
public async Task<IActionResult> Put([FromBody]UpdateEntityRequest request)
{
    await _mediator.Send(request); // Update entity
    var entite = await _mediator.Send(new GetByIdEntityRequest { Id = request.id }); // Get returned dto
    return Ok(entite);
}
```

Regards

**Comment by jbogard** [comment]
No, your request/responses are defined at one layer too deep. Request/response/handlers should be defined at the application level, where each API endpoint has a distinct request and handler. Maybe not a distinct response, as it is common for multiple API endpoints to return the same response.

**Reply by berubejp** [reply]
Thanks for your quick response.
> Maybe not a distinct response, as it is common for multiple API endpoints to return the same response.

I'm not sure to understand. If I am to return the same response for 3 endpoints, how can I achieve that? All I can think of is a base handler but isn't that against the idea of vertical slicing?

**Reply by jbogard** [reply]
Three different methods can return the same type without a base class. Inheritance is not the only way to share logic to create an object, there are many others. If that logic if needs to be shared in the first place.
